pkg/controller/watermarkpodautoscaler/replica_calculator.go

```diff
@@ -79,25 +93,20 @@ func (c *ReplicaCalculator) GetExternalMetricReplicas(logger logr.Logger, curren
 		value.Delete(prometheus.Labels{wpaNamePromLabel: wpa.Name, metricNamePromLabel: metricName})
 		return ReplicaCalculation{0, 0, time.Time{}}, fmt.Errorf("unable to get external metric %s/%s/%+v: %s", wpa.Namespace, metricName, selector, err)
 	}
-	logger.Info("Metrics from the External Metrics Provider", "metrics", metrics)
-
-	averaged := 1.0
-	if wpa.Spec.Algorithm == "average" {
-		averaged = float64(currentReplicas)
-	}
+	logger.V(4).Info("Metrics from the External Metrics Provider", "metrics", metrics)
 
 	var sum int64
 	for _, val := range metrics {
 		sum += val
 	}
-
-	replicaCount, utilizationQuantity := getReplicaCount(logger, currentReplicas, wpa, metricName, averaged, sum, metric.External.LowWatermark, metric.External.HighWatermark)
+	adjustedUsage := float64(sum) / averaged
```

Comment:
could you add comment to this operation?
I don't understand what means adjusted here. Is average can be equal to 0.0 ?